api/src/main/java/io/opencensus/trace/unsafe/ContextUtils.java

```diff
@@ -39,7 +37,34 @@ private ContextUtils() {}
    * The {@link io.grpc.Context.Key} used to interact with {@link io.grpc.Context}.
    *
    * @since 0.5
+   * @deprecated from API since 0.21. Use {@link #withValue(Context, Span)} and {@link
+   *     #getValue(Context)} instead.
+   */
+  // TODO(songy23): make this private once gRPC migrates to use the alternative APIs.
+  @Deprecated
+  public static final Context.Key<Span> CONTEXT_SPAN_KEY =
+      Context.<Span>keyWithDefault("opencensus-trace-span-key", BlankSpan.INSTANCE);
+
+  /**
+   * Creates a new {@code Context} with the given value set.
+   *
+   * @param context the parent {@code Context}.
+   * @param span the value to be set.
+   * @return a new context with the given value set.
+   * @since 0.21
+   */
+  public static Context withValue(Context context, Span span) {
+    return Utils.checkNotNull(context, "context").withValue(CONTEXT_SPAN_KEY, Utils.checkNotNull(span, "span"));
```

Comment:
no need to check for not null for the span, because we can support null Span correct?

Reply:
Fixed.